### PR TITLE
Turn License generator into a proper stack script

### PR DESCRIPTION
**Pull request by kRITZCREEK**

### CODE REVIEW

**Comment by garyb** [comment]
Do we need to add the dependency too, or can it figure that out?

**Comment by kRITZCREEK** [comment]
It does based on the documentation here: https://haskell-lang.org/tutorial/stack-script

In the stack user-guide it still says to add the packages https://docs.haskellstack.org/en/stable/GUIDE/#script-interpreter

whichever you prefer is fine with me

**Comment by kRITZCREEK** [comment]
@hdgarrood Good catch, I actually ran into the same and had just forgotten to apply the patch.

File: license-generator/generate.hs [hunk 1]
```diff
@@ -6,7 +9,7 @@
 --
 -- It is recommended to run this as follows:
 --
--- stack list-dependencies --flag purescript:RELEASE | stack exec runhaskell license-generator/generate.hs > LICENSE
+-- stack list-dependencies --flag purescript:RELEASE | stack license-generator/generate.hs > LICENSE
```
Comment (kRITZCREEK) [inline license-generator/generate.hs]:
```suggestion
-- stack ls dependencies --flag purescript:RELEASE | stack license-generator/generate.hs > LICENSE
```

**Comment by hdgarrood** [comment]
The appveyor failure is not relevant (and besides this PR doesn't touch anything which is exercised by the CI script anyway). Shall we merge this?

**Comment by kRITZCREEK** [comment]
Sounds good to me :)

**Pull request merged**